docs/docsite/rst/dev_guide/developing_modules_in_bulk.rst

```diff
@@ -0,0 +1,156 @@
+* Have a look at the existing modules, especially in the same functional area (such as cloud, networking, databases) in the :doc:`../list_of_all_modules` for a list of existing modules and how they've been named.
+* Shared code can be places into `lib/ansible/module_utils/`
+* Shared documentation, for example describing common arguments, can be placed in `lib/ansible/utils/module_docs_fragments/`
+* Although not required, unit and/or integration tests are strongly recomended. Unit tests are especially valuable when external resources (such as cloud or network devices) are required. For more information see `test/` and the `Testing Working Group <https://github.com/ansible/community/blob/master/MEETINGS.md>`_.
+
+Naming Convention
+`````````````````
+
+As you may have noticed when looking under `lib/ansible/modules/` we support up to two directories deep (but no deeper), e.g. `databases/mysql`. This is used to group files on disk as well as group related modules in the Module Index, for example: :doc:`../list_of_database_modules`.
+
+The directory name should represent the **product** or **OS** name, not the company name.
+
+Each module should have the above (or similar) prefix, see existing :doc:`../list_of_all_modules` for existing examples.
+
+**Note:**
+
+* File and directory names are always in lower case
+* Words are separated with an underscore (`_`) character
+* Module names should be in the singular, rather than plural, eg ``command`` not ``commands``
+
+
+Speak to us
+```````````
+
+Circulating your ideas before coding is a good way to help you set off in the right direction.
+
+After reading the "Before you start coding" section you will hopefully have a reasonable idea of the structure of your modules.
+
+We've found that writing a list of your proposed module names and a one or two line description of what they will achieve and having that reviewed by Ansible is a great way to ensure the modules fit the way people have used Ansible Modules before, and therefore make them easier to use.
+
+FIXME: How do we want to know the above, will everyone had a named contact at Ansible, what about if it's the community, rather than a vendor?
+
+FIXME: Some light weight review process of the above, maybe in IRC Core Meetings?
+
+
+Where to get support
+````````````````````
+Ansible has a very strong community, and we trust you as module developers will help grow.
+
+The community is a great pool of resource and knowledge.
+
+On :doc:`community` you can find how to:
+
+* Subscribe to the Mailing Lists - We suggest "Ansible Development List" (for codefreeze info) and "Ansible Announce list"
+* `#ansible-devel` - We have found that IRC `#ansible-devel` works best for module developers so we can have an interactive dialogue.
+* Join the various weekly IRC meetings
+
+
+Your First PR
+``````````````
+
+Assuming you've been through the rest of this document, and not just skipped ahead, you should now be ready to raise your first PR.
```

Comment:
Slightly scold-y.  :)  How about "Now that you've reviewed this document, you should be ready to open your first pull request."